lib/thrift/generator/enum_generator.ex

```diff
@@ -1,23 +1,47 @@
 defmodule Thrift.Generator.EnumGenerator do
+
   def generate(name, enum) do
     macro_defs = Enum.map(enum.values, fn {key, value} ->
-      macro_name = key |> to_string |> String.downcase |> String.to_atom
+      macro_name = to_ordinal(key)
       quote do
         defmacro unquote(Macro.var(macro_name, nil)), do: unquote(value)
       end
     end)
+
     member_defs = Enum.map(enum.values, fn {_key, value} ->
       quote do
         def member?(unquote(value)), do: true
       end
     end)
+
+    ordinal_defs = Enum.map(enum.values, fn {key, value} ->
+      ordinal_name = to_ordinal(key)
+      quote do
+        def ordinal(unquote(value)), do: unquote(ordinal_name)
+      end
+    end)
+
+    ordinals = enum.values
+    |> Keyword.keys
+    |> Enum.map(&to_ordinal/1)
+
     quote do
       defmodule unquote(name) do
         @moduledoc unquote("Auto-generated Thrift enum #{enum.name}")
         unquote_splicing(macro_defs)
+
+        unquote_splicing(ordinal_defs)
+        def ordinal(_), do: :"!!INVALID"
```

Comment:
i struggled with that, none of the options are good. I think nil might be the best.
 * There won't be a match error if the function isn't defined, it'll be a NoFunctionClauseError and it could be confusing.
 * raise -ing will almost certainly cause runtime errors that will be much more severe than they should be (oh, someone sent you a new value that you can't process for a single event? die.)
I'll switch to nil.